app/services/balance_transactions/updater.rb

```diff
@@ -0,0 +1,32 @@
+class BalanceTransactions::Updater
+  Args = Struct.new(:date, :comment, :amount, :transaction_id)
+
+  def initialize(args)
+    @args = Args.new(args[:date], args[:comment], args[:amount], args[:transaction_id])
+  end
+
+  def call
+    ActiveRecord::Base.transaction do
+      update_balance_transaction
+      update_transaction_data
+    end
+    @_transaction
+  end
+
+  private
+
+  def update_balance_transaction
+    transaction.transactinable.update(
+      comment: @args.comment,
+      date: @args.date
+    )
+  end
+
+  def transaction
+    @_transaction = Transaction.find(@args[:transaction_id])
+  end
+
+  def update_transaction_data
+    @_transaction.update(amount: @args.amount)
+  end
```

Comment:
@sinpai let's use "transaction" instead @_transaction or use @transaction for variable.
And move this method over `#transaction`.

Reply:
For me if code hasn't been read yet by interpreter - it shouldn't be after method where it was used. My opinion is the  opposite one. Let's wait for the @AleksSenkou last word :)

Reply:
@sinpai @marderer rails make many things for us - one of them is allowing us to place methods wherever we want. So, there is no need to place methods before a direct call, because the interpreter is pretty smart to create valid methods dependencies.

RoR allows us to write more humanized code. We can use it on our look.
According to that - we can try to write not just code, but code that people can read and understand quicker.

People reading from up to down, so let's write code up to down 😉(method definition comes after direct call)

Reply:
@sinpai @marderer what do you think? does it make sense?

Reply:
@sinpai @marderer what do you think? does it make sense?

Reply:
@sinpai if yes - let's refactor this PR with described practice

Reply:
I would refactor it according to this practice.